### ISSUE TITLE
Align manage tokens screen

### ISSUE THREAD

**Issue by tschubotz** [gnosis/safe#145]
There are still some difference in the way users enable and disable tokens.
Basically, the following features are necessary:

- Enable a token
- Hide a token
- Reorder enabled tokens (That would be the order on the assets overview
- Search for tokens

Android currently has [these screens](https://app.zeplin.io/project/5af063c9b4dc859b6bdaf897/dashboard?seid=5c7678b1d391d651e702a727)
iOS has [these screen](https://app.zeplin.io/project/5c49ce9866e3b3bee966f431/dashboard?seid=5c518a759c3e07307bfee601)

- I don't mind it being a little different if this makes sense due to OS specific things.
- Android currently has no way to reorder tokens. How do we imagine this to happen?
- We need an image for when we couldn't load the icon of a token for Android.

**Comment by biocom** [comment]
I created a proposal for Android. On some aspects it might differ in functionality from iOS, which I'd like to discuss so we can find consensus. The clickable prototype: https://invis.io/UMRAXF06542

**Notes:**
- Reorder action icon is shown upfront. Tap/hold on reorder icon to drag up/down. Saves the need to have a specific 'Edit' mode (and thus EDIT action on top right).
- Tap/hold to select 1 or more tokens to delete.
   - This specifically allows us to have a single + (add token) action in the right corner. 
   - While selecting is active, the left corner shows a back button, allowing you to cancel this action.
- Like iOS you can also swipe to the left to quickly delete a record.
- Vertical A-Z scroll menu like iOS. But hidden while search is active. @posthnikova You think we should also have it in search? What happens if there are only a couple of results (or no results) while having the vertical bar? It basically then has no practical use? Let me know your thoughts there.
- When adding a token you can add multiple tokens at once by enabling the checkbox. Wasn't sure from the iOS screens if this was also the case there @posthnikova ?
- When adding a token, the token list shows the tokens you already added with an active checkbox.

**Comment by tschubotz** [comment]
UX sync:
- Let's use the checkmarks

**Comment by posthnikova** [comment]
Changes after UX sync:

> Vertical A-Z scroll menu like iOS. But hidden while search is active. @posthnikova You think 
> we should also have it in search? What happens if there are only a couple of results (or no 
> results) while having the vertical bar? It basically then has no practical use? Let me know your 
> thoughts there.

Currently ios only shows found letters in search results and does not show alphabet when nothing is found. 
Corrected https://zpl.io/and8djQ, https://zpl.io/bAB9KPo 

> When adding a token you can add multiple tokens at once by enabling the checkbox. Wasn't 
> sure from the iOS screens if this was also the case there 

You can now check and uncheck multiple tokens: https://zpl.io/29o3orA

**Comment by biocom** [comment]
Thank you for the updates. As for a difference between Android/iOS is the way we render the individual token items. In my view having the token project name on a separate line, allows for a better overview, especially in terms of quickly scanning the token symbols from top to down. Should we align this layout or allow for it being a platform specific implementation?

<img width="337" alt="Screen Shot 2019-04-03 at 13 04 34" src="https://user-images.githubusercontent.com/31534717/55474322-1fd50e00-5611-11e9-9375-576b0f17c1ba.png">
<img width="360" alt="Screen Shot 2019-04-03 at 13 03 00" src="https://user-images.githubusercontent.com/31534717/55474324-1fd50e00-5611-11e9-8eb5-f4269394fde7.png">

**Comment by KristinaMayman** [comment]
I like having the two sets of information on different lines. If it's not a big concern that they take up more space this way, I would go for that option.

**Comment by biocom** [comment]
Also forgot to add:
- Adding the token project name, particularly when adding (new) tokens, can be helpful. Even users with sufficient knowledge about several token projects, would potentially find it useful in terms of identifying tokens unknown/not so known to them.

**Comment by posthnikova** [comment]
Version with two lines: 

https://invis.io/3HQ098MTCPU#/341237560_10_Add_Token

https://invis.io/3HQ098MTCPU#/341237556_07_Editing_Mode

**Comment by biocom** [comment]
@posthnikova Looks good to me.

**Comment by biocom** [comment]
After @posthnikova and I did another review of the Manage Tokens screen, we're thinking if it's better to keep the flow simpler than our latest proposals. 

Essentially having a single list where you can toggle on/off individual tokens (how the current main net Android app has it implemented) but with the search functionality. A summary:

- Use a single token list.
- Each token either is enabled or disabled.
- *You can search/scroll to fetch tokens outside the initial loaded tokens.
- Enabled tokens are always visible without fetching (local storage).
- Re-arranging the order of tokens will be taken out of the Manage Tokens and moved to the Assets overview screen (if technically feasible on both platforms). Could be marked as a 'low priority' feature.

*My understanding after discussing with @rmeissner is that we could go with a non-paginated list of tokens to not make this implementation (initially) too complicated. The enabled toggle state of a token is stored locally.

If in the future we we'd want to support pagination, we'd have to review if we can support that in combination with (locally) stored enabled token toggle states.

**Comment by biocom** [comment]
Added Android screens for the simplified approach:
https://invis.io/39RJTMLHUFS
https://invis.io/5MRJTOMQ24E
https://invis.io/AVRJTOVEF5G
https://invis.io/XMRJTP5V7N5

**Comment by sche** [comment]
@posthnikova @tschubotz IMO it does not make sense to align it in iOS app with the proposed scope for Android. When I discussed it with @rmeissner, a common understanding was that:
1) We need to align visual styles
2) Reordering on Android is not a priority, that's why Android can live with the current solution for the time being. On Android, there is no system reordering functionality like on iOS.
3) It is OK to fetch all available tokens, and pagination or top300 tokens is a discussion for future when we have this issue.
4) It is OK that some parts of the app are not 1 to 1 in terms of UX.

If we decide to have _somewhere_ in future one common UX for manage tokens with reordering on assets view page, or like iOS done it, or smth else, before we start redoing things that already work and fulfill the basic use case, we need a _complete_ design that will include all use cases:
1) reordering
2) pagination (if needed)
3) add a remote token
4) add custom token
4a) how to display it if a user disables it (should it be left in the list or not?)

I think it does not bring any value to make changes in the iOS app at the moment apart from aligning visual styles for the current solution, to consider the "design alignment" goal for Q1/Q2 to be achieved. The same for Android.

**Comment by tschubotz** [comment]
UX sync:
- Let keep it platform specific due to other more important things to do.
iOS:
- Keep the implemented version, just some UI update.

Android
- Let's add search + alphabet slider
- UI improvements
- Have the tokens sorted alphabetically (in manage tokens and on assets overview)

**Comment by biocom** [comment]
Updated the Android screens:
https://invis.io/XMS995P5NUA

Let me know if this is the direction we want/can implement for Android.

**Comment by posthnikova** [comment]
Visually looks good. Is adding custom token currently out of scope on android?

**Comment by tschubotz** [comment]
I think it looks good as well. Custom tokens is out of scope on mobile so far, yes.

**Comment by biocom** [comment]
Zeplin screens for Android are added:
https://app.zeplin.io/project/5af063c9b4dc859b6bdaf897/dashboard?seid=5c7678b1d391d651e702a727

**Comment by biocom** [comment]
I ran the checklist to close this ticket and have some comments:
- As for the manage tokens overview, we use different titles:
  - Android -> `Manage Tokens` -> https://zpl.io/aRqlmBe
  - iOS -----> `Tokens` ---------> https://zpl.io/beoxoZq
  I think Manage Tokens is a better title. But understand the space limitation on iOS. Keep the 
  difference?
- The search placeholder text is different:
   - Android -> `Search & select tokens...` -> https://zpl.io/bJKMpnE
   - iOS -----> `Search` ----------------------> https://zpl.io/VQK3Klm
  Again, keep the different text labels in place on both platforms?

**Comment by posthnikova** [comment]
@biocom 

• Menu item says "Manage Tokens", therefore it should be "Manage Tokens" https://zpl.io/aMK3KrK

• On Android it's both searching and selecting so I guess it should be different

**Comment by biocom** [comment]
@posthnikova Agreed. Thank you.
